### DISCUSSION TITLE
styleClass property of p-menuBar [model] list @Input works differently: migation documentation missing

### DISCUSSION THREAD

**Discussion by eXpertise7** [primefaces/community#162] [PrimeNG] [answered]
Here's a clear video how **`styleClass `of `p-menuBar` `[model]` list `@Input`** works **differently** in primeng v15.2.2 and in primeng v16.4.1.

https://www.youtube.com/watch?v=kkxRck0IWuI (watch in 720p quality)

In primeng 15 styleClass populates `< a >` element  CSS class, but in primeng 16 styleClass populates` < li >` element CSS class.

Don't you guys think there should be some migration documentation for stuff like these? Because maybe some of current users have dependent CSS logic like I do, and stuff breaks now because stuff work different.

I would not call this a bug, but I'd just suggest that migration documentation should be done to improve user satisfaction.

--
Issue is persistent on: PrimeNG - version 16.4.1, browser cache has been cleared by me after updating new version.

I'm available for more demonstration or explanation.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

We've updated the code structure of the menu components by implementing accessibility enhancements. Applying styleClass to the anchor elements wasn't a best practice and we've updated the structure. That's why the difference occurs between v15 and v16 all the changes are noted and users are notified in the changelog of the related version.
